docs/ci.md

```diff
@@ -54,3 +54,201 @@ Received 0 of 614674276 (0.0%), 0.0 MBs/sec
+[INFO] Listening for transport dt_socket at address: 18293
+[INFO] Listening for transport dt_socket at address: 53782
+```
+
+> This is rather inconvenient, and if someone finds a better way, please share! At the moment, this
+> means you'll have to test each port until you find the right one.
+
+To connect the debugger, you will need to port-forward the correct port to your local machine. The
+first step here is to set up your k8s credentials for the cluster:
+
+```shell
+gcloud container clusters get-credentials zeebe-ci --zone europe-west1-b --project zeebe-io
+```
+
+Next, switch to the right context and namespace:
+
+```shell
+kubectx gke_zeebe-io_europe-west1-b_zeebe-ci
+kubens default
+```
+
+We can now port-forward our debugger agent:
+
+```shell
+kubectl port-forward pod/actions-runner-16-n5tz8-nwtn2 5005:53782
+```
+
+> You can find the pod/runner name by going to your workflow run, selecting the job where the
+> test is running, and looking at the first section of the logs, i.e. `Set up job`. The runner name
+> is the pod name.
+
+The above will forward connections to `localhost:5005` on your machine to the remote node at
+port `53782`, where the agent is listening.
+
+> Unfortunately, the IntelliJ `Attach to process` shortcut will not work here, as it will not
+> discover the remote agent. You will have to create a new run configuration using
+> the `Remote JVM Debug` template, targeting `localhost:5005` (which should be the default).
+
+If you're running multiple tests/forks, and you need to figure out which port is the right one, I
+suggest catching the failing assertion error, and adding a busy loop there, with a breakpoint in the
+body. When connecting, it will stop there, and you will know you've connected to the right port. It
+will also prevent your test from continuing, allowing you to inspect the state. For example:
+
+```java
+try {
+  SslAssert.assertThat(socketAddress)
+    .as("node %s is not secured correctly at address %s", nodeId, address)
+    .isSecuredBy(certificate);
+} catch (final AssertionError e) {
+  final var watch = Stopwatch.createStarted();
+  while (watch.elapsed().toNanos() < Duration.ofHours(1).toNanos()) {
+    LockSupport.parkNanos(Duration.ofSeconds(1).toNanos());
+  }
+}
+```
+
+To cut down on reproduction times, you might sometimes want to run multiple workflow instances in
+parallel. There I can recommend using [Webhook.site](https://webhook.site/), and having your test
+ping that endpoint with some payload, so you can react to a test failure. For example:
+
+
+```java
+try {
+  SslAssert.assertThat(socketAddress)
+    .as("node %s is not secured correctly at address %s", nodeId, address)
+    .isSecuredBy(certificate);
+} catch (final AssertionError e) {
+  final var webhook = URI.create(
+      "https://webhook.site/<WEBHOOK UUID>?runnerId="
+    + System.getenv("RUNNER_NAME")
+    + "&pid="
+    + ProcessHandle.current().pid());
+  try {
+    HttpClient.newHttpClient()
```

Comment:
👍 that's a nice way of awaiting a failure instead of checking all the runs manually! "the test calls you :)" next level: slack DM 😆

Reply:
If you know of a one liner to send Slack DMs without secrets or adding a library, I'd prefer that over webhook :smile: